lib/algfld.gi

```diff
@@ -105,7 +105,7 @@ local fam,i,cof,red,rchar,impattr,deg;
   fam!.deg:=deg;
   i:=List([1..DegreeOfLaurentPolynomial(p)],i->fam!.zeroCoefficient);
   i[2]:=fam!.oneCoefficient;
-  i:=ImmutableVector(rchar,i,true);
+  i:=ImmutableVector(Size(f),i,true);
```

Comment:
Also: If you add `Fixes #1561` or `Resolves #1561` to the commit message (say as its last line), then merging this PR will automatically close #1561.